lib/http_aws_sigv4.c

```diff
@@ -120,214 +115,264 @@ CURLcode Curl_output_aws_sigv4(struct Curl_easy *data, bool proxy)
     return CURLE_OK;
   }
 
-  if(content_type) {
-    content_type = strchr(content_type, ':');
-    if(!content_type)
-      return CURLE_FAILED_INIT;
-    content_type++;
-    /* Skip whitespace now */
-    while(*content_type == ' ' || *content_type == '\t')
-      ++content_type;
+  /*
+   * Parameters parsing
+   * Google and Outscale use the same OSC or GOOG,
+   * but Amazon uses AWS and AMZ for header arguments.
+   * AWS is the default because most of non-amazon providers
+   * are still using aws:amz as a prefix.
+   */
+  tmp0 = data->set.str[STRING_AWS_SIGV4] ?
+    data->set.str[STRING_AWS_SIGV4] : "aws:amz";
+  tmp1 = strchr(tmp0, ':');
+  len = tmp1 ? tmp1 - tmp0 : strlen(tmp0);
```

Comment:
I got that warning multiple times:
```
http_aws_sigv4.c:167:20: warning: operand of ‘?:’ changes signedness from ‘long int’ to ‘size_t’ {aka ‘long unsigned int’} due to unsignedness of other operand [-Wsign-compare]
  167 |       len = tmp1 ? tmp1 - tmp0 : strlen(tmp0);
```
maybe `len = tmp1 ? (size_t)(tmp1 - tmp0) : strlen(tmp0);`